beekeeper-path-scheduler-apiary/src/main/java/com/expediagroup/beekeeper/scheduler/apiary/handler/MessageEventHandler.java

```diff
@@ -0,0 +1,147 @@
+/**
+ * Copyright (C) 2019-2020 Expedia, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.expediagroup.beekeeper.scheduler.apiary.handler;
+
+import java.time.Duration;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeParseException;
+import java.util.Collections;
+import java.util.EnumMap;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import com.expedia.apiary.extensions.receiver.common.event.ListenerEvent;
+import com.expedia.apiary.extensions.receiver.common.messaging.MessageEvent;
+
+import com.expediagroup.beekeeper.core.model.EntityHousekeepingPath;
+import com.expediagroup.beekeeper.core.model.HousekeepingPath;
+import com.expediagroup.beekeeper.core.model.LifecycleEventType;
+import com.expediagroup.beekeeper.core.model.PathStatus;
+import com.expediagroup.beekeeper.scheduler.apiary.filter.FilterType;
+import com.expediagroup.beekeeper.scheduler.apiary.filter.ListenerEventFilter;
+import com.expediagroup.beekeeper.scheduler.apiary.model.EventModel;
+
+public abstract class MessageEventHandler {
+
+  private static final Logger log = LoggerFactory.getLogger(MessageEventHandler.class);
+  private static final String CLIENT_ID = "apiary-metastore-event";
+  private final LifecycleEventType lifecycleEventType;
+  private final String cleanupDelay;
+  private final String hivePropertyKey;
+  private final List<FilterType> validFilterClasses;
+  @Autowired private EnumMap<FilterType, ListenerEventFilter> filterTypeMap;
```

Comment:
It feels weird to have this field autowired here, and not have this argument passed in from the subclass through the constructor. 

(we've had this discussion briefly before) If it was passed in from the subclass, it brings in to question the need for a mapping of  `ListenerEventFilter` and type altogether. The enums feel a bit superfluous as the classes themselves are pretty explicit. The list of valid filters is currently located in the subclass anyway, so this could just be a list of classes (the filters themselves are stateless, so we could remove the Spring bean annotations from them, and create them explicitly quite easily).

@massdosage any thoughts?

Reply:
agreed - the filter logic & sharing that logic between handlers via the parent hasn't been the cleanest solution. I think that if we are okay w/ having the filters be explicitly instantiated in the child handler class, that should simplify the setup and answer a couple of the other open questions I had about filters

Reply:
Yeah I'm happy with either, if we have to maintain a list of filters for a handler, the least amount of code to do so would be nice. Let's see what others think...

Reply:
@max-jacobs @massdosage Updated this so that each handler is required to instantiate their own filters. This removes the enum & weird mapping to get this working correctly